magpie/ui/management/views.py

```diff
@@ -171,58 +188,62 @@ def view_users(self):
-                       u"too_long_user_name": False, u"form_user_name": u"", u"form_user_email": u"",
-                       u"user_groups": self.get_all_groups(first_default_group=users_group)}
-        check_data = [u"conflict_group_name", u"conflict_user_name", u"conflict_email",
-                      u"invalid_user_name", u"invalid_email", u"invalid_password"]
+        return_data = {"conflict_group_name": False, "conflict_user_name": False, "conflict_user_email": False,
+                       "invalid_user_name": False, "invalid_user_email": False, "invalid_password": False,
+                       "mismatch_password": False, "too_long_user_name": False,
+                       "form_user_name": "", "form_user_email": "",
+                       "user_groups": self.get_all_groups(first_default_group=users_group)}
+        check_data = ["conflict_group_name", "conflict_user_name", "conflict_email",
+                      "invalid_user_name", "invalid_email", "invalid_password", "mismatch_password"]
 
         if "create" in self.request.POST:
             groups = self.get_all_groups()
             user_name = self.request.POST.get("user_name")
             group_name = self.request.POST.get("group_name")
             user_email = self.request.POST.get("email")
             password = self.request.POST.get("password")
-            return_data[u"form_user_name"] = user_name
-            return_data[u"form_user_email"] = user_email
+            confirm = self.request.POST.get("confirm")
+            return_data["form_user_name"] = user_name
+            return_data["form_user_email"] = user_email
 
             if group_name not in groups:
-                data = {u"group_name": group_name}
+                data = {"group_name": group_name}
                 resp = request_api(self.request, schemas.GroupsAPI.path, "POST", data=data)
                 if resp.status_code == HTTPConflict.code:
-                    return_data[u"conflict_group_name"] = True
+                    return_data["conflict_group_name"] = True
             if user_email in self.get_user_emails():
-                return_data[u"conflict_user_email"] = True
+                return_data["conflict_user_email"] = True
             if user_email == "":
-                return_data[u"invalid_user_email"] = True
+                return_data["invalid_user_email"] = True
             if len(user_name) > get_constant("MAGPIE_USER_NAME_MAX_LENGTH"):
-                return_data[u"too_long_user_name"] = True
+                return_data["too_long_user_name"] = True
             if user_name in self.get_user_names():
-                return_data[u"conflict_user_name"] = True
+                return_data["conflict_user_name"] = True
             if user_name == "":
-                return_data[u"invalid_user_name"] = True
+                return_data["invalid_user_name"] = True
             if password is None or isinstance(password, six.string_types) and len(password) < 1:
```

Comment:
Don't we have a parameter for the password length (rather than checking against < 1). Maybe that's why I got a json error when creating a user with a bad password rather than a smooth html invalid password warning.

Reply:
e64159a now returns the explicit "Invalid 'password' ... (<12 characters)"  message

Reply:
The account self editing password functionnality returns a generic 500 error when setting a password less than 12 characters rather than showing an explicit message like with the new user form. (Same bad behavior when using the admin password editing)
![image](https://user-images.githubusercontent.com/4202336/93219920-e5462f00-f739-11ea-976d-2c5e20f0d511.png)